packages/headless/src/features/commerce/context/context-actions.ts

```diff
@@ -0,0 +1,71 @@
+import { createAction } from '@reduxjs/toolkit';
+import {nonEmptyString, requiredNonEmptyString, validatePayload} from '../../../utils/validate-payload';
+import {UserParams, ViewParams} from '../../../api/commerce/commerce-api-params';
+import { RecordValue } from '@coveo/bueno';
+
+export interface SetContextPayload {
+  trackingId?: string;
+  language?: string;
+  currency?: string;
+  clientId?: string;
+  user?: UserParams;
+  view: ViewParams;
+}
+
+export const setContext = createAction(
+  'commerce/setContext',
+  (payload: SetContextPayload) =>
+    validatePayload(payload, {
```

Comment:
Similar to my above comment: This complex schema seems a copy of this of what's already in headless-context.ts.